File: Python/array/first_missing_positive.py
from typing import List

def firstMissingPositive(nums: List[int]) -> int:
    n = len(nums)
    for i in range(n):
        if nums[i] <= 0:
            nums[i] = n + 1
    
    for i in range(n):
        if abs(nums[i]) < n + 1:
            nums[abs(nums[i]) - 1] = -abs(nums[abs(nums[i]) - 1])

    for i in range(n):
        if nums[i] > 0:
            return i + 1
    
    return n + 1

if __name__ == "__main__":
    print(firstMissingPositive([1,1]))

File: core/tmp/Java/src/com/dbc/algorithm/Array/FirstMissingPositive.java
package com.dbc.algorithm.Array;

import java.util.Arrays;

public class FirstMissingPositive {
    public static int firstMissingPositive(int[] nums) {
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] <= 0) {
                nums[i] = nums.length + 1;
            }
        }

        System.out.println(Arrays.toString(nums));

        for (int i = 0; i < nums.length; i++) {
            if (nums[i] < nums.length + 1) {
                nums[Math.abs(nums[i]) - 1] = -Math.abs(nums[Math.abs(nums[i]) - 1]);
            }
        }

        System.out.println(Arrays.toString(nums));

        for (int i = 0; i < nums.length; i++) {
            if (nums[i] > 0) {
                return i + 1;
            }
        }
        return nums.length + 1;
    }
}
